### PR TITLE
Channels and goroutines

### PR DESCRIPTION
**Pull request by vrecan**
using channels + go routines. Need to look into how to manage shutdown but otherwise good

### CODE REVIEW

File: diskStat/diskStat.go [hunk 1]
```diff
@@ -2,7 +2,7 @@ package diskStat
 import (
    "strconv"
    )
-type diskStat struct {
+type DiskStat struct {
```
Comment (weberr13) [inline diskStat/diskStat.go:5]: I'm ok with heads down camel case if we want to use it, my concern here is that the filename is now inconsistent.

File: iostat.go [hunk 2]
```diff
@@ -5,30 +5,41 @@ import (
    "bufio"
    "log"
    "strings"
+   "time"
    "./diskStat"
```
Comment (weberr13) [inline iostat.go:9]: can this be removed?  How does go handle multiple imports of the same package?

File: iostat.go [hunk 3]
```diff
@@ -5,30 +5,41 @@ import (
...
+    for scanner.Scan() {
+    	// fmt.Println(scanner.Text())
+    	line := strings.Fields(scanner.Text())
+    	stat,err := diskStat.LineToStat(line)
+    	if(nil != err) {
+    		log.Fatal(err)
+    	}
+    	my_channel <- stat
+    }
+    if err := scanner.Err(); err != nil {
```
Comment (weberr13) [inline iostat.go:40]: this is pretty cool

**Comment by weberr13** [comment]
:+1:

File: diskStat/diskStat.go (hunk 1 above)
Comment (vrecan) [inline diskStat/diskStat.go:5]: The capital letter has meaning, it makes it public

File: iostat.go (hunk 2 above)
Comment (vrecan) [inline iostat.go:9]: it deals with dup's automatically, it won't compile if I remove it

**Pull request merged**